scripts/deleteSnapsets.js

```diff
@@ -0,0 +1,162 @@
+            // docsToRemove and have them removed.
+            // Note - if there is a way to determine the final "end" event, then
+            // it could  call doBatchDelete().
+            resp.on("end", function () {
+                var gpiiKeyRecords = JSON.parse(respString);
+                fluid.each(gpiiKeyRecords.rows, function (gpiiKey) {
+                    fluid.log("GPII Key: " + gpiiKey.value._id);
+                    dbLoader.deleteGpiiKey(gpiiKey.value);
+                });
+            });
+        });
+        getGpiiKeysRequest.on("error", function (e) {
+            fluid.log("Error finding snapsets' associated GPII Keys: " + e.message);
+        });
+        getGpiiKeysRequest.end();
+    });
+    dbLoader.doBatchDelete(docsToRemove);
+};
+
+/**
+ * Delete the given GPII Key record.
+ * @param {Object} gpiiKey  - The key to delete.
+ */
+dbLoader.deleteGpiiKey = function (gpiiKey) {
+    var deleteOptions = {
+        hostname: dbLoader.parsedCouchDbUrl.hostname,
+        port: dbLoader.parsedCouchDbUrl.port,
+        path: "",           // filled in below.
+        method: "DELETE",
+        headers: {
+            "Accept": "application/json"
+        }
+    };
+    deleteOptions.path = "/gpii/" + gpiiKey._id + "?rev=" + gpiiKey._rev;
+    var deleteGpiiKeyRequest = http.request(deleteOptions, function (res) {
+        fluid.log("STATUS: " + res.statusCode);
+        fluid.log("HEADERS: " + JSON.stringify(res.headers, null, 2));
+    });
+    deleteGpiiKeyRequest.on("error", function (e) {
+        fluid.log("Error finding GPII Key: " + e.message);
+    });
+    deleteGpiiKeyRequest.end();
+};
+
+/**
+ * Delete the snapset Prefs Safes and their associated GPII Keys.
+ * @param {Array} docsToRemove  - Array of records to delete.
+ */
+dbLoader.doBatchDelete = function (docsToRemove) {
+    var batchDeleteOptions = {
+        hostname: dbLoader.parsedCouchDbUrl.hostname,
+        port: dbLoader.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // filled in below
+            "Content-Type": "application/json"
+        }
+    };
+    var batchPostData = JSON.stringify({"docs": docsToRemove});
+    batchDeleteOptions.headers["Content-Length"] = Buffer.byteLength(batchPostData);
+    var batchDeleteReq = http.request(batchDeleteOptions, function (res) {
+        fluid.log("STATUS: " + res.statusCode);
+        fluid.log("HEADERS: " + JSON.stringify(res.headers, null, 2));
+        res.on('end', function () {
+            fluid.log('Batch deletion of snapsets');
+        });
+    });
+    batchDeleteReq.on('error', function (e) {
+        fluid.log("Error deleting snapset Prefs Safes: " + e.message);
+    });
+    batchDeleteReq.write(batchPostData);
+    batchDeleteReq.end();
+};
+
+dbLoader.snapSetsRequest = http.request(dbLoader.prefsSafesViewUrl, dbLoader.processSnapsets);
```

Comment:
The couchdb url on the GPII cloud might be using https instead http. Will find out by testing with the developer cloud.

Reply:
Any progress on this, @cindyli ?

Reply:
Sorry about the confusion. I meant YOU will find out when testing with the cloud, not me. It's also worth to check with @mrtyler on this information since production/staging might act differently than dev clusters.